### PR TITLE
Make `--event-time-start` and `--event-time-end` mutually required

### LINKED ISSUE #10874: [Feature] Make `--event-end-time` require `--event-start-time` and vice versa
Currently, we have _five_ ways to run a microbatch model 🥴 
1. creating/recreating the model
    * batches are run from model's defined `begin` to "now"
2. incrementally with no extra args
    * the batches run are determined by the current time and the lookback
    * f.x. if you have a `batch_size=day`, `lookback=2`, and it is "currently" `2024-10-17 13:04:39` then three batches are run: `2024-10-15`, `2024-10-16`, and `2024-10-17`
3. incrementally with both `--event-time-start` and `--event-time-end` specified
    * batches are run >= `--event-time-start` and < `--event-time-end`
    * f.x. if you have a `batch_size=day`, `--event-time-start "2024-10-01"`, and `--event-time-end "2024-10-04"` then three batches will be run: `2024-10-01`, `2024-10-02`, `2024-10-03`
4. incrementally with _only_ `--event-time-start` specified
    * batches will be run from `--event-time-start` to "now"
    * f.x. if you have a `batch_size=day`, `--event-time-start "2024-10-13"`, and it's currently midday on 2024-10-17, then five batches will be run: `2024-10-13`, `2024-10-14`, `2024-10-15`, `2024-10-16`, and `2024-10-17`
5. incrementally with _only_ `--event-time-end` specified
    * batches will run the `--event-time-end` batch plus prior batches according to the `period`
    * f.x. if you have a `batch_size=day`, `lookback=2`, and `--event-time-end "2024-10-04"`, then three batches will be run: `2024-10-01`, `2024-10-02`, `2024-10-03`

Through further discussion, we believe (4) and (5) are confusing, unexpected, and dangerous. Ways (4) and (5) weren't intentional implementations, but accidental realities created by implementations for (1), (2), and (3). As such, we should _require_ that the specification of `--event-time-end` necessitates `--event-time-start` and vice versa.

### PR DESCRIPTION
**Pull request by QMalcolm**
Resolves #10874 

### Problem

`--event-time-start` and `--event-time-end` were not mutually required, which led to some weird/bad behavior if only one was specified

### Solution

Make it so that one being specified necessitates the other.

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me.
- [X] I have run this code in development, and it appears to resolve the stated issue.
- [X] This PR includes tests, or tests are not required or relevant for this PR.
- [X] This PR has no interface changes (e.g., macros, CLI, logs, JSON artifacts, config files, adapter interface, etc.) or this PR has already received feedback and approval from Product or DX.
- [X] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions.

### CODE REVIEW

File: core/dbt/cli/flags.py [hunk 1]
```diff
@@ -362,17 +361,26 @@ def _validate_event_time_configs(self) -> None:
             getattr(self, "EVENT_TIME_END") if hasattr(self, "EVENT_TIME_END") else None
         )
 
-        if event_time_start is not None and event_time_end is not None:
```
Comment (QMalcolm) [inline core/dbt/cli/flags.py:365]: It's probably easier to see what changed if viewing commit by commit. Git didn't handle displaying the changes made across the 2 commits very gracefully.

**Review by gshank — APPROVED** [review]

**Comment by mirnawong1** [comment]
hey @QMalcolm , sorry for the add'l question but do we need to update the docs here? https://docs.getdbt.com/docs/build/incremental-microbatch#timezones

**Comment by QMalcolm** [comment]
> hey @QMalcolm , sorry for the add'l question but do we need to update the docs here? https://docs.getdbt.com/docs/build/incremental-microbatch#timezones

@mirnawong1 Questions are great! No need to apologize 🙂 If the question is in reference to timezones (the section linked) then no update is needed as UTC will still be assumed. If the question is in reference to the page overall, yes we should probably update the document somewhere to note that `--event-time-start` and `--event-time-end` are mutually necessary (i.e. the specification of one requires the specification of the other).

**Comment by mirnawong1** [comment]
docs pr here to address this: https://github.com/dbt-labs/docs.getdbt.com/pull/6351/files

**Review by MichelleArk — APPROVED** [review]

**Pull request merged**